pilot/cmd/pilot-agent/main.go

```diff
@@ -78,13 +78,14 @@ var (
 	loggingOptions         = log.DefaultOptions()
 	outlierLogPath         string
 
-	instanceIPVar        = env.RegisterStringVar("INSTANCE_IP", "", "")
-	podNameVar           = env.RegisterStringVar("POD_NAME", "", "")
-	podNamespaceVar      = env.RegisterStringVar("POD_NAMESPACE", "", "")
-	kubeAppProberNameVar = env.RegisterStringVar(status.KubeAppProberEnvName, "", "")
-	serviceAccountVar    = env.RegisterStringVar("SERVICE_ACCOUNT", "", "Name of service account")
-	clusterIDVar         = env.RegisterStringVar("ISTIO_META_CLUSTER_ID", "", "")
-	callCredentials      = env.RegisterBoolVar("CALL_CREDENTIALS", false, "Use JWT directly instead of MTLS")
+	instanceIPVar               = env.RegisterStringVar("INSTANCE_IP", "", "")
+	podNameVar                  = env.RegisterStringVar("POD_NAME", "", "")
+	podNamespaceVar             = env.RegisterStringVar("POD_NAMESPACE", "", "")
+	kubeAppProberNameVar        = env.RegisterStringVar(status.KubeAppProberEnvName, "", "")
+	serviceAccountVar           = env.RegisterStringVar("SERVICE_ACCOUNT", "", "Name of service account")
+	clusterIDVar                = env.RegisterStringVar("ISTIO_META_CLUSTER_ID", "", "")
+	callCredentials             = env.RegisterBoolVar("CALL_CREDENTIALS", false, "Use JWT directly instead of MTLS")
+	xdsEnableTokenFetchExchange = env.RegisterBoolVar("XDS_ENABLE_TOKEN_FETCH_EXCHANGE", false, "Enable token fetch and exchange for XDS")
```

Comment:
Can we use XDS server address to figure out if token exchange is needed?

Reply:
This proposal has been discussed in other comments of this PR. Quoted from comments of @qfel: "".googleapis.com" for less corner cases; But it's not quite OK to hardcode that - for local testing, this could be localhost or some in-K8s proxy."

Reply:
Can we use a simpler and cleaner name ? 'token fetch exchange' is not very intuitive for users. 

I would also like to see all settings related to STS in the same place ( we have one for stackdriver, one for meshCA - and now this ).

Reply:
How about: XDS_AUTH="xxx" - where the value can be 'gcp' for the GCP-specific auth exchange ( what this PR is actually implementing), etc.
We can eventually switch CA_AUTH to same style - this will allow other forms of vendor-specific auth schemes. Combined with making the TokenProvider an interface - or using the interface provided by gRPC library ( which already has several other implementations ) we can reduce the confusion.

Reply:
Renamed to XDS_AUTH_PROVIDER="xxx" where the value can be 'gcp'. 
"XDS_AUTH" is already taken by an env variable in pilot/pkg/features/pilot.go.